spec/services/sources/create_service_spec.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Sources::CreateService do
+  describe '#call' do
+    let(:organisation) { create(:organisation) }
+
+    context 'when with unnormalized url' do
+      it 'normalizes the url' do
+        source = described_class.call(name: 'Source One',
+                                      url: 'https://localhost/api/foo/../?')
+        expect(source.url).to eq('https://localhost/api/')
```

Comment:
Good thinking with normalising the urls, very nice! That said I'm not sure I understand why `/foo` is being stripped out here, but not `/api`?

Reply:
Because it works as a unix directory:
```
➜  ~ pwd
/Users/ngw
➜  ~ mkdir api
➜  ~ mkdir api/foo
➜  ~ cd api/foo/..
➜  api pwd
/Users/ngw/api```

Reply:
Ahh I see! Thank you 😌